src/ui/public/state_management/state_monitor.js

```diff
@@ -8,9 +8,14 @@ function stateMonitor(state, defaultState) {
   let destroyed = false;
   let ignoredProps = [];
   let changeHandlers = [];
+  let originalState;
 
-  // state.toJSON returns a reference, clone so we can mutate it safely
-  const originalState = cloneDeep(defaultState) || cloneDeep(state.toJSON());
+  setOriginalState(defaultState);
+
+  function setOriginalState(defaultState) {
```

Comment:
I'm getting a little mixed up here, because `defaultState` and `originalState` seem to be pretty much the same thing, with different names. Can we use consistent terminology here? Either `default` or `original`?